src/hypothesis/strategies.py

```diff
@@ -648,13 +659,23 @@ def characters(whitelist_categories=None, blacklist_categories=None,
                 min_codepoint, max_codepoint
             )
         )
+    if (
+        whitelist_characters is not None and
+        min_codepoint is None and max_codepoint is None and
+        whitelist_categories is None and blacklist_categories is None
```

Comment:
I think this might be clearer with `all` - either `all(arg is None for arg in (...))` or as individual clauses, but it feels like too many ands.